src/index.ts

```diff
@@ -690,6 +717,180 @@ class Spanner extends GrpcService {
+   * @param {google.longrunning.Operation} operation An operation object that
+   *     can be used to check the status of the request.
+   * @param {google.longrunning.IOperation} apiResponse The full API response.
+   */
+  /**
+   * Create an instance config.
+   *
+   * Wrapper around {@link v1.InstanceAdminClient#createInstanceConfig}.
+   *
+   * @see {@link v1.InstanceAdminClient#createInstanceConfig}
+   * @see [CreateInstanceConfig API Documentation](https://cloud.google.com/spanner/docs/reference/rpc/google.spanner.admin.instance.v1#google.spanner.admin.instance.v1.InstanceAdmin.CreateInstanceConfig)
+   *
+   * @throws {GoogleError} If a name is not provided.
+   * @throws {GoogleError} If a configuration object is not provided.
+   * @throws {GoogleError} If a base config is not provided in the configuration
+   *                       object.
+   *
+   * @param {string} name The name of the instance config to be created.
+   * @param {CreateInstanceConfigRequest} config Configuration object.
+   * @param {CreateInstanceConfigCallback} [callback] Callback function.
+   * @returns {Promise<CreateInstanceConfigResponse>}
+   *
+   * @example
+   * ```
+   * const {Spanner} = require('@google-cloud/spanner');
+   * const spanner = new Spanner();
+   *
+   * const [baseInstanceConfig] =
+   *     await spanner.getInstanceConfig(baseInstanceConfigId);
+   * const config = {
+   *   baseConfig: baseInstanceConfig.name,
+   *   replicas: baseInstanceConfig.replicas.concat(baseInstanceConfig.optionalReplicas[0])
+   * };
+   *
+   * function callback(err, instance, operation, apiResponse) {
+   *   if (err) {
+   *     // Error handling omitted.
+   *   }
+   *
+   *   operation
+   *     .on('error', function(err) {})
+   *     .on('complete', function() {
+   *       // Instance created successfully.
+   *     });
+   * }
+   *
+   * spanner.createInstanceConfig('custom-new-instance-config', config, callback);
+   *
+   * //-
+   * // If the callback is omitted, we'll return a Promise.
+   * //-
+   * spanner.createInstanceConfig('custom-new-instance-config', config)
+   *   .then(function(data) {
+   *     const instanceConfig = data[0];
+   *     const operation = data[1];
+   *     return operation.promise();
+   *   })
+   *   .then(function() {
+   *     // Instance config created successfully.
+   *   });
+   * ```
+   */
+  createInstanceConfig(
+    name: string,
+    config: CreateInstanceConfigRequest
+  ): Promise<CreateInstanceConfigResponse>;
+  createInstanceConfig(
+    name: string,
+    config: CreateInstanceConfigRequest,
+    callback: CreateInstanceConfigCallback
+  ): void;
+  createInstanceConfig(
+    name: string,
+    config: CreateInstanceConfigRequest,
+    callback?: CreateInstanceConfigCallback
+  ): void | Promise<CreateInstanceConfigResponse> {
+    if (!name) {
+      throw new GoogleError('A name is required to create an instance config.');
```

Comment:
I don't remember the latest status on errors, so just to be sure: Is this consistent with the other errors that we return in this client library?

Reply:
Yes, it is consistent with other errors that we return in this client library. [This](https://github.com/googleapis/nodejs-spanner/pull/1554) is the PR where we made it consistent across the handwritten layer.